### PR TITLE
coponents/form: add `FormFieldOptionGroup` component

### PR DESCRIPTION
**Pull request by maceksimon**
Add new component to show selectable options on `RegisterChallengePage`.

Component is not being used yet. It will become part of the second form step.
Preview in test.

### CODE REVIEW

File: src/components/__tests__/FormFieldOptionGroup.cy.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import { colors } from 'quasar';
+import FormFieldTestWrapper from 'components/global/FormFieldTestWrapper.vue';
+import { i18n } from '../../boot/i18n';
+import { rideToWorkByBikeConfig } from '../../boot/global_vars';
+import { hexToRgb } from 'app/test/cypress/utils';
```
Comment (tmszi) [inline src/components/__tests__/FormFieldOptionGroup.cy.js:5]: We can replace this custom `hexToRgb` function with the Quasar `hexToRgb` function https://quasar.dev/quasar-utils/color-utils#introduction throughout the code if possible. You can do this in the next new PR.

File: src/components/form/FormFieldOptionGroup.vue [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+      set(value: string) {
+        emit('update:modelValue', value);
+      },
+    });
+
+    const { isFilled } = useValidation();
+
+    const options = [
+      {
+        label: 'S kolegy z práce',
```
Comment (tmszi) [inline src/components/form/FormFieldOptionGroup.vue:72]: Each string must be translatable.

File: src/components/form/FormFieldOptionGroup.vue [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+import { defineComponent, computed } from 'vue';
+
+// composables
+import { useValidation } from 'src/composables/useValidation';
+
+// types
+import { ConfigGlobal } from '../types';
+
+// config
+const rideToWorkByBikeConfig: ConfigGlobal = JSON.parse(
```
Comment (tmszi) [inline src/components/form/FormFieldOptionGroup.vue:37]: Here boot/global_vars.ts we have declared and parsed global config. Do you really need to parse the JSON configuration here again?

https://github.com/auto-mat/ride-to-work-by-bike-frontend/blob/a52a67d85e78a1f847ccf58e7165cb297797ab3f/src/boot/global_vars.ts#L8-L10

File: src/components/form/FormFieldOptionGroup.vue [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        return props.modelValue;
+      },
+      set(value: string) {
+        emit('update:modelValue', value);
+      },
+    });
+
+    const { isFilled } = useValidation();
+
+    const options = [
```
Comment (tmszi) [inline src/components/form/FormFieldOptionGroup.vue:70]: Missing type?

File: src/components/__tests__/FormFieldOptionGroup.cy.js (hunk 1 above)
Comment (maceksimon) [reply src/components/__tests__/FormFieldOptionGroup.cy.js:5]: Updated. However, the function unfortunately does not return string, but an object.
So additional rewrite into rgb(xxx) value is needed to work in the test.
`const colorPrimary = 'rgb(${rgbPrimary.r}, ${rgbPrimary.g}, ${rgbPrimary.b})';`

File: src/components/form/FormFieldOptionGroup.vue (hunk 2 above)
Comment (maceksimon) [reply src/components/form/FormFieldOptionGroup.vue]: Fixed.

File: src/components/form/FormFieldOptionGroup.vue (hunk 3 above)
Comment (maceksimon) [reply src/components/form/FormFieldOptionGroup.vue:38]: No, you are correct. We can import directly from `import { rideToWorkByBikeConfig } from 'src/boot/global_vars';`

File: src/components/form/FormFieldOptionGroup.vue (hunk 4 above)
Comment (maceksimon) [reply src/components/form/FormFieldOptionGroup.vue]: Fixed, making use of the updated `FormOption` type.

**Review by tmszi — APPROVED** [review]

**Pull request merged**